pinot-common/src/main/java/com/linkedin/pinot/common/query/QueryRequest.java

```diff
@@ -30,24 +37,88 @@
 public class QueryRequest {
   private static final Logger LOGGER = LoggerFactory.getLogger(QueryRequest.class);
 
-  private String clientId = "";
   private InstanceRequest instanceRequest;
+  // Future to track result of query execution
+  private ListenableFuture<DataTable> resultFuture;
+  // Timing information for different phases of query execution
+  private final TimerContext timerContext;
+
+  // executor service to parallelize query tasks
+  private ListeningExecutorService queryWorkers;
+
+  private final ServerMetrics serverMetrics;
 
-  public QueryRequest(InstanceRequest request) {
+  public QueryRequest(InstanceRequest request, ServerMetrics serverMetrics) {
     this.instanceRequest = request;
+    this.serverMetrics = serverMetrics;
+    BrokerRequest brokerRequest = (request == null) ? null : request.getQuery();
+    timerContext = new TimerContext(brokerRequest, serverMetrics);
   }
 
-  public String getClientId() {
-    return clientId;
+  public @Nullable String getClientId() {
```

Comment:
changed to getBrokerId()
Initial thinking was to keep it more generic (for non-broker clients sending requests) but that can be confusing